Solution1/BusinessLayer/Vlidators/DataForWeatherRequestValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using BusinessLayer.DTOs;
+using FluentValidation;
+using System;
+using System.Configuration;
+
+namespace BusinessLayer.Vlidators
+{
+    public class DataForWeatherRequestValidator : AbstractValidator<DataForWeatherRequestDTO>
+    {
+        public DataForWeatherRequestValidator()
+        {
+            RuleSet("CityName", () =>
+            {
+                RuleFor(p => p.CityName)
+                    .NotEmpty()
```

Comment:
not empty, MaximumLength have standard error messages, you shouldn't specify WithMessage